src/engine/strat_engine/backstore/cleanup.rs

```diff
@@ -0,0 +1,30 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// Code to handle cleanup after a failed operation.
+
+use super::super::super::errors::{EngineError, EngineResult, ErrorEnum};
+
+use super::blockdev::StratBlockDev;
+
+/// Wipe some blockdevs of their identifying headers.
+/// Return an error if any of the blockdevs could not be wiped.
+/// If an error occurs while wiping a blockdev, attempt to wipe all remaining.
+pub fn wipe_blockdevs(blockdevs: &[StratBlockDev]) -> EngineResult<()> {
+    let mut unerased_devnodes = Vec::new();
+
+    for bd in blockdevs {
+        let bd_devnode = bd.devnode.to_owned();
+        bd.wipe_metadata()
+            .unwrap_or_else(|_| unerased_devnodes.push(bd_devnode));
+    }
+
```

Comment:
Would this be more rustic?

```rust
blockdevs
        .iter()
        .for_each(|bd| {
                      let bd_devnode = bd.devnode.to_owned();
                      bd.wipe_metadata()
                          .unwrap_or_else(|_| unerased_devnodes.push(bd_devnode))
                  });
```

Reply:
Sort of, but there is some feeling that a for is better when the whole purpose is not to compute something but to have the side-effect. So we generally stick with the for-loop. Also, I just moved this code, I didn't work on it in this PR, so I'm going to leave it as is.